src/mame/nec/pce_m.cpp

```diff
@@ -23,32 +23,41 @@ void pce_state::init_tg16()
 
 void pce_state::machine_start()
 {
+	if (m_cartslot->exists())
+	{
+		// Install HuCard ROM
+		if ((m_cartslot->get_type() != PCE_ACARD_DUO))
+		{
+			m_maincpu->space(AS_PROGRAM).install_readwrite_handler(0x000000, 0x0fffff, read8sm_delegate(m_cartslot, FUNC(pce_cart_slot_device::read_cart)), write8sm_delegate(m_cartslot, FUNC(pce_cart_slot_device::write_cart)));
+		}
+
+		// Install Super System Card registers
+		if ((m_cartslot->get_type() == PCE_CDSYS3J)
+			|| (m_cartslot->get_type() == PCE_CDSYS3U)
+			|| (m_cartslot->get_type() == PCE_ACARD_PRO))
+		{
+			m_maincpu->space(AS_PROGRAM).install_read_handler(0x1ff8c0, 0x1ff8c7, 0, 0x330, 0, read8sm_delegate(m_cartslot, FUNC(pce_cart_slot_device::read_ex)));
+		}
+
+		// Install Arcade Card registers
+		if ((m_cartslot->get_type() == PCE_ACARD_DUO)
+			|| (m_cartslot->get_type() == PCE_ACARD_PRO))
+		{
+			m_maincpu->space(AS_PROGRAM).install_readwrite_handler(0x080000, 0x087fff, read8sm_delegate(m_cartslot, FUNC(pce_cart_slot_device::read_ram)), write8sm_delegate(m_cartslot, FUNC(pce_cart_slot_device::write_ram)));
+			m_maincpu->space(AS_PROGRAM).install_readwrite_handler(0x1ffa00, 0x1ffbff, read8sm_delegate(m_cartslot, FUNC(pce_cart_slot_device::peripheral_r)), write8sm_delegate(m_cartslot, FUNC(pce_cart_slot_device::peripheral_w)));
+		}
+	}
+
```

Comment:
This is hella leaky, the HuCard slot can only access up to `0x0fffff`, the rest belongs to expansion bus which afaik can map itself anywhere in the upper bank.
I feel like moving stuff like this is not much better than the old code ...

Reply:
CPU A20 pin is tied into XCE pin of HuCard slot, so HuCard can accessible to over ~`0x100000` area.
reference: https://gamesx.com/wiki/doku.php?id=hu6280_netlist&s[]=hucard, [TurboGrafx-16 Unit Service Manual](https://gamesx.com/wiki/lib/exe/fetch.php?media=schematics:turbografx-16_unit_service_manual_-_smtg16.pdf)

Reply:
So it's a direct overlay, and my point still stand: this initialization should be local to the cart bus, not handled by the pce_m.cpp (which is a deprecated file bucket that should have stuff removed, not added in any way or form).